### DISCUSSION TITLE
Using Expo packages like Expo Secure Store in widgets/native code

### DISCUSSION THREAD

**Discussion by lapto** [expo/expo#33689] [Archived: Questions] [answered]
Hello.

I have an expo app with a widget. I would like for them to be able to share secure data.
On react native side I can use the "Expo secure store", but on native side I cant as the function to store/get data is inside the Module file (https://github.com/expo/expo/blob/main/packages/expo-secure-store/ios/SecureStoreModule.swift)

Its written as a **final class**, and all the method beside the definition is of type **private** . Is there any reason why the package is so locked down? Is there any way to use these method on native side? 

It seems like a waste to have to make a native module and more or less copy the implementation just so it can be used from within a widget.

**Comment by tsapeta** [comment, marked as answer]
Hey!

We usually start from non-public access levels when there is no need for it to be public. If something is public, changing it may introduce breaking changes for some users so it requires us to be more careful. 

We're open to expose those as `public`/`open`. Moreover, I think we should also move these functions to a separate class and make them static, so you can use them from anywhere in the native code, without having to search for the module instance in the module registry. Would you like to help us doing so?

**Reply by wcastand** [reply]
i would like to help doing that, been looking at it but can't figure it out. How do i make static method in a class available to other modules.

been trying to look at expo module core which has a Utilities public static class but can't manage to make it work for expo-secure-store.

any pointers would be appreciated to try to do that :)

My use case for it is i'm working on a module for PassKit, to add a card to apple wallet app but i need to go back and forth between swift and typescript with a ref to the button to call some backend api, i'd like to avoid that by doing the api call from swift but for that i need the user token stored in the expo-secure-store :)

**Reply by wcastand** [reply]
Any chanced someone could point me  in the right direction to make expo  secure store available on native side? i'd like to use it for widget and share the user token with the native side, i don't like the idea of putting the token in  the group shared data when it's already in the secure store.
